specreduce/background.py

```diff
@@ -87,6 +88,13 @@ def _to_trace(trace):
         bkg_wimage = np.zeros_like(self.image, dtype=np.float64)
         for trace in self.traces:
             trace = _to_trace(trace)
+            if (np.any(trace.trace.data >= self.image.shape[self.crossdisp_axis]) or
+                    np.any(trace.trace.data < 0)):
+                raise ValueError("center of background window goes beyond image boundaries")
```

Comment:
Should an error only be raised when `all` (instead of `any`) of the center of the background window is out of bounds? I assume `any` triggering an error instead of a warning could be annoying in a situation where only a couple of values are off-frame due to a wiggly input trace.

If we switched to `all`, we'd have to confirm that any trace values passed to `_ap_weight_image()` here are less than `self.image.shape[self.crossdisp_axis] + hwidth` in order to avoid an `IndexError` from `_get_boxcar_weights()`.

Reply:
That is an option.  The current implementation raises a warning if any of any window goes out of bounds, but an error if the _center_ of any window goes out of bounds (meaning that at least half of at least one of the windows extends out-of-bounds).  My concerns about extending this logic so the error is only raised if all of the window(s) is out-of-bounds is that the same warning will then be raised for 1-pixel out-of-bounds as only 1-pixel remaining.  And if we want to change the error to if no pixels at all are included from any of the windows, then the logic does become significantly more complex.

Reply:
If I'm understanding correctly that this branch's approach is actually more forgiving than `main`'s, that's good to know. I am interested in what others think about this topic but don't think it should delay a merge much longer if we don't hear back.